### ISSUE TITLE
Change pie chart color 

### ISSUE THREAD

**Issue by hypolas** [go-analyze/charts#14]
Hello,

I try to set color to pie chart serie.

In my case, I need, for exemple, to set color "red" (for error data) and green (for valide data) .

That seems don't work.  I try with:
```
	p, err := charts.PieRender(
		values,
		charts.TitleOptionFunc(charts.TitleOption{
			Text:    "Rainfall vs Evaporation",
			Subtext: "Fake Data",
			Left:    charts.PositionCenter,
		}),
		charts.PaddingOptionFunc(charts.Box{
			Top:    20,
			Right:  20,
			Bottom: 10,
			Left:   20,
		}),
		charts.LegendOptionFunc(charts.LegendOption{
			Orient: charts.OrientVertical,
			Data: []string{
				"tests",
				"failures",
				"errors",
			},
			Left: charts.PositionLeft,
		}),
		charts.PieSeriesShowLabel(),
		func(opt *charts.ChartOption) {
			opt.SeriesList[1].Style.FillColor = drawing.ColorRed
			opt.SeriesList[2].Style.FillColor = drawing.ColorRed
		},
	)
```

**Comment by jentfoo** [comment]
Thank you for opening this issue @hypolas.  This fork has been re-evaluating the chart configuration options and the series `Style` has not yet been considered.

Looking at the implementation, it looks like setting the `FillColor` currently only functions on Bar charts.  This could be easily extended to also support Pie Charts, but there ends up being a weird side effect that the Legend does not have a reference to the series (also true for bar charts), and thus it's colors are not updated, for example:
![TestPieChartSeriesStyleFill-actual-2925364196](https://github.com/user-attachments/assets/27ae0c27-95c3-4cf0-9b0e-47059ff5699f)

One goal of this fork is to make themes [more straight forward](https://github.com/go-analyze/charts/pull/4).  We don't require themes to be globally registered (they are passed in with the chart configuration).  Would a suitable alternative be to take a theme you like, and then replace the `SeriesColors` with the colors you want represented?  I still want to make theme mutations easier, it would look something like this:
```
	baseTheme := charts.GetTheme(ThemeVividLight)
	var seriesColors []charts.Color
	for i := range series {
		if red {
			seriesColors = append(seriesColors, drawing.ColorRed)
		} else {
			seriesColors = append(seriesColors, baseTheme.GetSeriesColor(i))
		}
	}
	customTheme := baseTheme.WithSeriesColors(seriesColors)
	// or possibly
	customTheme := baseTheme.WithOptions(ThemeOption{SeriesColors: seriesColors})
```

Then your pie chart would be constructed like this:
```
	p, err := charts.PieRender(
		values,
		ThemeOptionFunc(customTheme),
		charts.TitleOptionFunc(charts.TitleOption{
			Text:    "Rainfall vs Evaporation",
			Subtext: "Fake Data",
			Left:    charts.PositionCenter,
		}),
		charts.PaddingOptionFunc(charts.Box{
			Top:    20,
			Right:  20,
			Bottom: 10,
			Left:   20,
		}),
		charts.LegendOptionFunc(charts.LegendOption{
			Orient: charts.OrientVertical,
			Data: []string{
				"tests",
				"failures",
				"errors",
			},
			Left: charts.PositionLeft,
		}),
		charts.PieSeriesShowLabel(),
	)
```

I believe that having a single method of setting the theme for the series would be more straight forward, but I would appreciate your feedback before we make any changes.  Thank you!

**Comment by jentfoo** [comment]
@hypolas, I just submitted a PR which removes the series `Style` you were configuring and instead introduces the ergonomics around changing the series colors of the theme: https://github.com/go-analyze/charts/pull/15

Let me know your thoughts on moving forward with a change like this, or if you think there is a good reason to leave this style configuration at the series level.  Thank you!

**Comment by hypolas** [comment]
Hello.

Than you for your very fast answer.

For solve my problem quikly, I had modified my code like this :

```
func (g Graph) buildGraph() {
	p, err := charts.PieRender(
		g.Values,
		g.makeDefaultTheme(),
		charts.TitleOptionFunc(charts.TitleOption{
			Text:    g.Text,
			Subtext: g.Subtext,
			Left:    charts.PositionCenter,
		}),
		charts.PaddingOptionFunc(charts.Box{
			Top:    20,
			Right:  20,
			Bottom: 20,
			Left:   20,
		}),
		charts.LegendOptionFunc(charts.LegendOption{
			Orient: charts.OrientVertical,
			Data:   g.Legends,
			Top:    "5%",
			Left:   charts.PositionLeft,
		}),
		charts.PieSeriesShowLabel(),
	)

	if err != nil {
		panic(err)
	}

	if buf, err := p.Bytes(); err != nil {
		panic(err)
	} else if err = writeFile(buf); err != nil {
		panic(err)
	}
}

func (g Graph) makeDefaultTheme() charts.OptionFunc {
	th := charts.ThemeOptionFunc(charts.MakeTheme(charts.ThemeOption{
		SeriesColors:    g.Colors,
		TextColor:       drawing.ColorFromHex("555555"),
		BackgroundColor: drawing.ColorWhite,
	}))

	return th
}
````
g.Color is a slice of color previously generated with this code.

```
func colorSwitch(result string) drawing.Color {
	switch result {
	case "failed":
		return drawing.ColorRed
	case "failure":
		return drawing.ColorRed
	case "passed":
		return drawing.ColorLime
	case "ok":
		return drawing.ColorLime
	case "errors":
		colorError := drawing.Color{
			R: 244,
			G: 100,
			B: 100,
			A: 255,
		}

		return colorError
	}
	return drawing.ColorWhite
}
```

If this can help somebody ;)

**Comment by jentfoo** [comment]
Thank you for the feedback @hypolas, I am glad to hear that the custom theme options is a workable solution for you.

I went ahead and polished PR #15 further, making sure that Font configuration (which is broadly used and common) is encapsulated in the `FontStyle` struct instead of needing the broader `Style` struct (which could lead to similar confusion as you had here).  Feel free to give the PR a look if you have interest, I would appreciate any feedback you might have.

Once merged I plan to release this under version `0.2.0`.  Although you have a good solution here, the primary benefit to you will be the [WithSeriesColors](https://github.com/go-analyze/charts/pull/15/files#diff-979f3d4694bff55e81c3653fdeb8995f40372052bc15418158f98137a4653842R45) which could allow you to easily only change the series colors.

Let me know if you have questions or thoughts, thank you again!

**Comment by jentfoo** [comment]
These changes were just pushed under the `v0.2.0` tag.  Let me know if you have any questions or further recommendations.  Thank you!
